functions/install/java.sh

```
@@ -37,34 +37,111 @@ function install_java {
-        echo -e "\e[38;2;129;170;254m[INFO] \e[38;5;250mJava $JAVA_VERSION_S installed successfully.\e[0m"
+        printout info "Java $JAVA_VERSION_S installed successfully."
     fi
     export JAVA_HOME="$HOME/.sdkman/candidates/java/$JAVA_VERSION_S"
     export PATH="$JAVA_HOME/bin:$PATH"
 }
+
+# install_paper downloads the Paper server JAR (using MCJARS_API_KEY if set), creates the Paper server configuration, assigns a port, ensures Java is installed, and launches the server.
+function install_paper {
+    printout info "Downloading Paper Server..."
+    if [ -n "$MCJARS_API_KEY" ]; then
+        jar_url=$(curl --silent --request GET --header "Authorization: $MCJARS_API_KEY" --url "https://versions.mcjars.app/api/v2/builds/PAPER/$paper" | jq -r '.builds[0].jarUrl')
```

Shellcheck (0.11.0)

[warning] paper is referenced but not assigned.

(SC2154)